### ISSUE TITLE
Sourcing from large repos

### ISSUE THREAD

**Issue by jasoncodes** [freshshell/fresh#94]
Right now there’s no efficient way to source contrib files from large repositories such as [git](https://github.com/git/git). Examples include [git-completion.bash](https://github.com/git/git/blob/master/contrib/completion/git-completion.bash) and [diff-highlight](https://github.com/git/git/blob/master/contrib/diff-highlight/diff-highlight)

We should use GitHub’s raw URLs to retrieve these down in a single request.

We’ll need some way to tell fresh that it should not clone the whole repo. An additional option (e.g. `--no-repo`) or special source repo format (e.g. `git/git:raw`) seems like the most obvious way. I’d prefer to not have something GitHub specific if we decide to add a new option as it may be useful for other sources later as well. Ideally this should also work for _any_ http/https URLs. Another possible option is to set a flag for an entire source via a new `fresh-source` line.

Suggestions for how to trigger this new mode are welcome.

**Comment by jasoncodes** [comment]
I would also like a shorthand for GitHub like `fresh git/git contrib/completion/git-completion.bash --github-raw` but I’m not sure on what to call the option.

**Comment by twe4ked** [comment]
Maybe if we're using a GitHub source we can just use `--raw` as the option?

**Comment by jasoncodes** [comment]
Yeah, `--raw` is fine. I was hoping to find something that meant “use a fast access method for a single file rather than cloning the repo”.

**Comment by twe4ked** [comment]
Ah, that is harder.

`--use-a-fast-access-method-for-a-single-file-rather-than-cloning-the-repo`.
